File: Python_Library/Book.py
from Media import Media

class Book(Media):
	def __init__(self,a,b,c,d,e,f,g,h,i,j):
		Media.__init__(self,a,b,c,j)
		self.author=d
		self.description=e
		self.publisher=f
		self.city=g
		self.year=h
		self.series=i
		
	def display(self):
		print("<------------------------------ Record From Book --------------------------------------------->\n");
		print("Call Number: "+self.call_number+"\n");
		print("Title: "+self.title+"\n");
		print("Subjects: "+self.subject+"\n");
		print("Author: "+self.author+"\n");
		print("Description: "+self.description+"\n");
		print("Publisher: "+self.publisher+"\n");
		print("City: "+self.city+"\n");
		print("Year: "+self.year+"\n");
		print("Series: "+self.series+"\n");
		print("Notes: "+self.notes+"\n");
		
	def compare_other(self,x):
		if (self.description.find(x) != -1):
			return True
		elif(self.year.find(x) !=-1):
			return True
		elif(self.notes.find(x) !=-1):
			return True
		else:
			return False
	
 
  

File: ObjectOrientedDesign(LibrarySearch)/Book.cpp
#include<iostream>
#include<string>
#include"Book.h"
#include"Media.h"
using namespace std;


Book :: Book( string a,string b,string c,string d, string e, string f, string g, string h, string i,string j) : Media(a, b, c, j)
{  
  this->author=d;
  this->description=e;
  this->publisher=f;
  this->city=g;
  this->year=h;
  this->series=i;
  
}

void Book::display() const
{  
  cout<<"<------------------------------ Record From Book --------------------------------------------->"<<endl; 
  cout<<"Call Number:"<<call_number<<"\n"<<"Title:"<<title<<"\n"<<"Subjects:"<<subject<<"Author:"<<author<<"\n"<<"Description:"<<description<<
  "\n"<<"Publisher:"<<publisher<<"\n"<<"City:"<<city<<"\n"<<"Year:"<<year<<"\n"<<"Series:"<<series<<"\n"<<"Notes:"<<notes<<endl;
}


bool Book::compare_other(const string& x) const
{
  if (this->description.find(x) != std::string::npos)
  {
     return true;
  } 
  else if(this->year.find(x) != std::string::npos) 
  {
     return true;
  }
  else if(this->notes.find(x) != std::string::npos) 
  {
     return true;
  }
  else 
  {
     return false;
  }
}


